### ISSUE TITLE
[telemetry] Span and counters per system migration in SystemMigrationHelper

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#49]
## Goal

Emit a ph-telemetry span and success/failure counter per system migration in `ph-oton-mgrs`.

A migration that silently fails at startup is one of the worst failure modes an application has; right now the only evidence is an `ERROR` log line.

## Where

`ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/sysmigration/SystemMigrationHelper.java`

Two overloads, both needing the same treatment:

```java
public static void performMigrationIfNecessary (@NonNull ISystemMigrationManager aSysMigMgr,
                                                @NonNull @Nonempty String sMigrationID,
                                                @NonNull Runnable aMigrationAction)

public static void performMigrationIfNecessary (@NonNull ISystemMigrationManager aSysMigMgr,
                                                @NonNull @Nonempty String sMigrationID,
                                                @NonNull Supplier <SuccessWithValue <String>> aMigrationAction)
```

Both already log `"Performing migration '<id>'"` / `"Finished performing migration '<id>'"` and both already `catch (final RuntimeException ex)` and log `"Error execution system migration '<id>'"`. Neither declares a checked exception, so `Telemetry.withSpanVoid (...)` is enough — no throwing variant needed.

The second overload additionally distinguishes a *business* failure (`SuccessWithValue` reporting failure) from a *technical* one (a thrown `RuntimeException`). Both must be counted, and they must be distinguishable.

## What to emit

New `com.helger.photon.mgrs.CMgrsTelemetry`. Note that `CLongRunningJobTelemetry` stays exactly where it is — it has been public API since 10.4.0 and must not be moved or merged.

| Constant | Value |
|---|---|
| `SPAN_MIGRATION_EXECUTE` | `photon.migration.execute` |
| `METRIC_MIGRATIONS_EXECUTED` | `photon.migration.executed` |
| `METRIC_MIGRATION_DURATION` | `photon.migration.duration` |
| `ATTR_MIGRATION_ID` | `photon.migration.id` |
| `ATTR_MIGRATION_SUCCESS` | `photon.migration.success` |
| `ATTR_MIGRATION_FAILURE_KIND` | `photon.migration.failure.kind` (`none` / `business` / `technical`) |
| `UNIT_MIGRATION` | `{migration}` |
| `UNIT_MILLIS` | `ms` |

- **Span** `photon.migration.execute`, kind `INTERNAL`, attribute `photon.migration.id`. Set the status to `ERROR` for the business-failure case too — `withSpanVoid` only sees thrown exceptions and would otherwise mark a failed migration as OK.
- **Counter** `photon.migration.executed`, attributes `id`, `success`, `failure.kind`.
- **Histogram** `photon.migration.duration` in ms, attribute `id`.

The migration is skipped entirely when it has already run (`performMigrationIfNecessary` only acts if necessary). Emit **nothing** on the skip path — a counter that fires on every startup for every historical migration is noise.

## Cardinality

Migration IDs are string literals in application source code. They are bounded and grow slowly (one per release at most), so `photon.migration.id` is safe as a metric attribute — this is the rare case where an ID *is* an acceptable dimension. Say so explicitly in its Javadoc, so the next person does not copy the pattern to an unbounded ID.

## POM

None. `ph-oton-mgrs` already depends on `com.helger.telemetry:ph-telemetry`.

## Acceptance criteria

- [ ] Both overloads instrumented.
- [ ] Business failure and technical failure are distinguishable via `photon.migration.failure.kind`, and both mark the span as `ERROR`.
- [ ] Nothing is emitted when the migration is skipped as already-performed.
- [ ] Unit test alongside the existing `ph-oton-mgrs/src/test/java/.../sysmigration/SystemMigrationManagerTest.java`, using recording SPIs installed via `Telemetry.install (...)` / `TelemetryMetrics.install (...)` and reset to `null` in `@After`.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.6.0**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-mgrs/.../mgrs/CMgrsTelemetry.java` | new, public - span / metric / attribute names |
| `ph-oton-mgrs/.../sysmigration/SystemMigrationMetrics.java` | new, public - counter + histogram |
| `ph-oton-mgrs/.../sysmigration/SystemMigrationTelemetry.java` | new, package-private emitter |
| `ph-oton-mgrs/.../sysmigration/SystemMigrationHelper.java` | modified - both overloads |
| `ph-oton-mgrs/src/test/.../SystemMigrationTelemetryTest.java` | new, 4 tests |

No POM change - ph-oton-mgrs already depends on `ph-telemetry`. `CLongRunningJobTelemetry` is untouched and stays exactly where it is.

### Emission

* Span `photon.migration.execute` (`INTERNAL`), attribute `photon.migration.id`.
* Counter `photon.migration.executed` with `id`, `success` and `failure.kind`.
* Histogram `photon.migration.duration` with `id` only.

### Business vs technical failure

The second overload distinguishes them and both mark the span as failed:

| Outcome | `photon.migration.failure.kind` | Span status |
|---|---|---|
| Migration completed | `none` | OK |
| `SuccessWithValue` reported a failure | `business` | ERROR |
| The migration threw | `technical` | ERROR, with the exception recorded |

`Telemetry.withSpanVoid (...)` only ever sees thrown exceptions, and `SystemMigrationHelper` catches those itself - so **both** failure branches set the span status explicitly rather than relying on the helper. Covered by `testBusinessFailure` and `testTechnicalFailure`.

The failure kind is carried out of the span lambda in a `Wrapper <String>`, so the `finally` block that emits the metrics sees it.

### Nothing on the skip path

The whole instrumentation sits inside the existing `if (!aSysMigMgr.wasMigrationExecutedSuccessfully (sMigrationID))`, so a migration that already ran emits no span, no counter and no histogram. `testAlreadyPerformedMigrationEmitsNothing` asserts that both the tracer and the meter stay completely empty - a counter firing on every startup for every historical migration would be pure noise.

### Cardinality

`photon.migration.id` is a metric attribute, which is the exception to the usual rule. Its Javadoc says why - migration IDs are string literals in application source, bounded and growing at most one per release - and warns explicitly not to copy the pattern to an unbounded ID.

### Verification

`SystemMigrationTelemetryTest`: 4/4 green (success, skip, technical failure, business failure), using recording SPIs installed via `Telemetry.install (...)` / `TelemetryMetrics.install (...)`. That works here because no other ph-oton-mgrs test calls `performMigrationIfNecessary (...)`, so `SystemMigrationMetrics` is not class-loaded before the test installs its meter - unlike in ph-oton-security and ph-oton-audit. Full reactor `mvn install` over all 23 modules: BUILD SUCCESS. `README.md` news bullet added under `v10.6.0 - work in progress`.
